lesson_17/headers/custom_string.hpp

```diff
@@ -0,0 +1,93 @@
+#pragma once
+
+#define _CRT_SECURE_NO_WARNINGS
+#include <iostream>
+#include <cstring>
+
+
+class StringCustom {
+public:
+    StringCustom();
+    StringCustom& operator=(const char* c_str);
+    void assign(const char* c_str);
+
+    ~StringCustom() { delete[] m_str; } // ~StringCustom() { delete(?) m_str; }
+
+    std::size_t getLen() const { return strlen(m_str); }
+    std::size_t getSize() const { return getLen() + 1; }
+private:
+    char* m_str{ nullptr }; // char* m_str;
+
+};
+
+StringCustom::StringCustom()
+{
+    m_str = new char[1];
+    m_str[0] = '\0';
+}
+
+StringCustom& StringCustom::operator=(const char* c_str)
+{
+    delete[] m_str; m_str = nullptr;// MEMORY LEAK
+    if (c_str == nullptr)
+    {
+        m_str = new char[1]; //m_str = new char(?);
+        *m_str = '\0';
+    }
+    else
+    {
+        const std::size_t size = strlen(c_str);
```

Comment:
it should be `strlen(c_str) + 1;` tto reserve place for '\0' symbol

Reply:
done